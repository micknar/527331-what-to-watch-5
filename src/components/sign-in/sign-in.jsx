import React from "react";
import PropTypes from "prop-types";
import ReactLoading from 'react-loading';
import {connect} from "react-redux";
import PageHeaderLogo from "../page-header-logo/page-header-logo";
import PageFooter from "../page-footer/page-footer";
import {login} from "../../store/api-actions";
import {setIsLoginDataSending} from "../../store/action";

const getFieldClassName = (isValid) => isValid ? `sign-in__field` : `sign-in__field sign-in__field--error`;

const SignIn = (props) => {
  const {
    email,
    password,
    sendData,
    onInputChange,
    onFormSubmit,
    isValidEmail,
    isValidPassword,
    isLoginDataSending,
    isLoginError
  } = props;

  const onSignInBtnClick = (evt) => {
    evt.preventDefault();

    sendData({
      login: email,
      password,
    });
  };

  const getMessage = () => {
    if (isLoginDataSending) {
      return (
        <div
          className="sign-in__message"
          style={{display: `flex`, justifyContent: `center`, marginBottom: `20px`}}
        >
          <ReactLoading type={`bars`} color={`#dfcf77`} height={`10%`} width={`10%`} />
        </div>
      );
    } else if (!isValidEmail) {
      return (
        <div className="sign-in__message" >
          <p>Please enter a valid email address</p>
        </div>
      );
    } else if (!isValidPassword) {
      return (
        <div className="sign-in__message" >
          <p>Please enter a valid password</p>
        </div>
      );
    } else if (isLoginError) {
      return (
        <div className="sign-in__message" >
          <p>We canâ€™t recognize this email</p>
          <p>and password combination. Please try again.</p>
        </div>
      );
    } else {
      return ``;
    }
  };

  return (
    <div className="user-page">
      <header className="page-header user-page__head">
        <PageHeaderLogo />

        <h1 className="page-title user-page__title">Sign in</h1>
      </header>

      <div className="sign-in user-page__content">
        <form
          action="#"
          onSubmit={onFormSubmit}
          className="sign-in__form"
        >
          {getMessage()}
          <div className="sign-in__fields">
            <div className={getFieldClassName(isValidEmail)}>
              <input
                onChange={onInputChange}
                className="sign-in__input"
                type="email"
                placeholder="Email address"
                name="user-email"
                id="user-email"
              />
              <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
            </div>
            <div className={getFieldClassName(isValidPassword)}>
              <input
                onChange={onInputChange}
                className="sign-in__input"
                type="password"
                placeholder="Password"
                name="user-password"
                id="user-password"
              />
              <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
            </div>
          </div>
          <div className="sign-in__submit">
            <button
              className="sign-in__btn"
              type="submit"
              disabled={!isValidPassword}
              onClick={onSignInBtnClick}
            >
              Sign in
            </button>
          </div>
        </form>
      </div>

      <PageFooter />
    </div>
  );
};

SignIn.propTypes = {
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  sendData: PropTypes.func.isRequired,
  onInputChange: PropTypes.func.isRequired,
  onFormSubmit: PropTypes.func.isRequired,
  isValidEmail: PropTypes.bool.isRequired,
  isValidPassword: PropTypes.bool.isRequired,
  isLoginDataSending: PropTypes.bool.isRequired,
  isLoginError: PropTypes.bool.isRequired,
};

const mapStateToProps = ({USER}) => ({
  isLoginDataSending: USER.isLoginDataSending,
  isLoginError: USER.isLoginError,
});

const mapDispatchToProps = (dispatch) => ({
  sendData(authData) {
    dispatch(setIsLoginDataSending(true));
    dispatch(login(authData));
  }
});

export {SignIn};
export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
