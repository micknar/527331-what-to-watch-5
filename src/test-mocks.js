import {getDate} from "./utils";

export const noop = () => { };

export const films = [
  {
    id: 1,
    name: `Bronson`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `/img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 8.9,
    ratingMark: `Booring`,
    scoresCount: 240,
    director: `Nicolas Winding Refn`,
    starring: [`Bill Murray`, `Saoirse Ronan`],
    runTime: {
      hours: 2,
      minutes: 40,
    },
    genre: `Crime`,
    released: 1996,
    isFavorite: false,
  },
  {
    id: 2,
    name: `Midnight Special`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Midnight_Special.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 10,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 42,
    },
    genre: `Thriller`,
    released: 1989,
    isFavorite: true,
  },
  {
    id: 3,
    name: `Pulp Fiction`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Pulp_Fiction.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amo eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 2.2,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 22,
    },
    genre: `Action`,
    released: 2007,
    isFavorite: false,
  },
  {
    id: 4,
    name: `Matrix`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Matrix.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Matrix.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ext enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 4.5,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 24,
    },
    genre: `Adventure`,
    released: 2011,
    isFavorite: true,
  },
  {
    id: 5,
    name: `Legend`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Legend.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem i ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 4.1,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 57,
    },
    genre: `Comedy`,
    released: 2015,
    isFavorite: false,
  },
  {
    id: 6,
    name: `angs of new york`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/gangs_of_new_york.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sedrem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.uip ex ea commodo consequat.`,
    rating: 7.8,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 35,
    },
    genre: `Fantasy`,
    released: 1980,
    isFavorite: false,
  },
  {
    id: 7,
    name: `A Star Is Born`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/A_Star_is_Born.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_is_Born.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 6.4,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Guy Ritchie`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 25,
    },
    genre: `Drama`,
    released: 1989,
    isFavorite: false,
  },
  {
    id: 8,
    name: `Snatch`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Snatch.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliqliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 9.8,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 2,
      minutes: 20,
    },
    genre: `Fantasy`,
    released: 2004,
    isFavorite: false,
  },
  {
    id: 9,
    name: `Bohemian Rhapsody`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Bohemian_Rhapsody.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Bohemian_Rhapsody.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur.`,
    rating: 3,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: {
      hours: 1,
      minutes: 20,
    },
    genre: `Adventure`,
    released: 1998,
    isFavorite: false,
  },
];

export const comments = [
  {
    user: {
      id: 19,
      name: `Christina`,
    },
    rating: 2.4,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam`,
    date: getDate(`2020-10-09T13:38:44.769Z`),
  },
  {
    user: {
      id: 14,
      name: `Corey`,
    },
    rating: 3.2,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-27T13:38:44.769Z`),
  },
  {
    user: {
      id: 2,
      name: `Mollie`,
    },
    rating: 1.7,
    comment: `Lorem ipsustrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-09T13:38:44.769Z`),
  },
  {
    user: {
      id: 14,
      name: `Corey`,
    },
    rating: 4.2,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ilaboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-11T13:38:44.769Z`),
  },
  {
    user: {
      id: 11,
      name: `Jack`,
    },
    rating: 4.5,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitatboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-19T13:38:44.769Z`),
  },
];
