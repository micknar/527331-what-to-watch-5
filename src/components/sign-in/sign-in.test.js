import React from "react";
import renderer from "react-test-renderer";
import {SignIn} from "./sign-in";
import {noop} from "../../test-mocks";

jest.mock(`../page-header-logo/page-header-logo`, () => `PageHeaderLogo`);
jest.mock(`../page-footer/page-footer`, () => `PageFooter`);

describe(`Should SignIn render correctly`, () => {
  it(`With success authorization`, () => {
    const tree = renderer
      .create(
          <SignIn
            email={`mrrobot@gmail.com`}
            password={`helloworld`}
            sendData={noop}
            onInputChange={noop}
            onFormSubmit={noop}
            isValidEmail={true}
            isValidPassword={true}
            isLoginDataSending={false}
            isLoginError={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With valid data sending`, () => {
    const tree = renderer
      .create(
          <SignIn
            email={`mrrobot@gmail.com`}
            password={`helloworld`}
            sendData={noop}
            onInputChange={noop}
            onFormSubmit={noop}
            isValidEmail={true}
            isValidPassword={true}
            isLoginDataSending={true}
            isLoginError={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With invalid email`, () => {
    const tree = renderer
      .create(
          <SignIn
            email={`mrrobot@@@gmail.com`}
            password={`helloworld`}
            sendData={noop}
            onInputChange={noop}
            onFormSubmit={noop}
            isValidEmail={false}
            isValidPassword={true}
            isLoginDataSending={false}
            isLoginError={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With invalid password`, () => {
    const tree = renderer
      .create(
          <SignIn
            email={`mrrobot@gmail.com`}
            password={``}
            sendData={noop}
            onInputChange={noop}
            onFormSubmit={noop}
            isValidEmail={true}
            isValidPassword={false}
            isLoginDataSending={false}
            isLoginError={false}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With login error`, () => {
    const tree = renderer
      .create(
          <SignIn
            email={``}
            password={``}
            sendData={noop}
            onInputChange={noop}
            onFormSubmit={noop}
            isValidEmail={false}
            isValidPassword={false}
            isLoginDataSending={false}
            isLoginError={true}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
