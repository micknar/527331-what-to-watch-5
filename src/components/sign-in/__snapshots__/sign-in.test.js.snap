// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should SignIn render correctly With invalid email 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <PageHeaderLogo />
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          Please enter a valid email address
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field sign-in__field--error"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <PageFooter />
</div>
`;

exports[`Should SignIn render correctly With invalid password 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <PageHeaderLogo />
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          Please enter a valid password
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field sign-in__field--error"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <PageFooter />
</div>
`;

exports[`Should SignIn render correctly With login error 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <PageHeaderLogo />
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
      >
        <p>
          Please enter a valid email address
        </p>
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field sign-in__field--error"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field sign-in__field--error"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <PageFooter />
</div>
`;

exports[`Should SignIn render correctly With success authorization 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <PageHeaderLogo />
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <PageFooter />
</div>
`;

exports[`Should SignIn render correctly With valid data sending 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <PageHeaderLogo />
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__message"
        style={
          Object {
            "display": "flex",
            "justifyContent": "center",
            "marginBottom": "20px",
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path transform=\\"translate(2)\\" d=\\"M0 12 V20 H4 V12z\\"> 
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(8)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.2\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(14)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.4\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(20)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(26)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.8\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
</svg>
",
            }
          }
          style={
            Object {
              "fill": "#dfcf77",
              "height": "10%",
              "width": "10%",
            }
          }
        />
      </div>
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
            type="email"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <PageFooter />
</div>
`;
